src/recipes/tooling-engine-singleton.md

```diff
@@ -0,0 +1,119 @@
+<!--
+  ~ Copyright (c) godot-rust; Bromeon and contributors.
+  ~ This Source Code Form is subject to the terms of the Mozilla Public
+  ~ License, v. 2.0. If a copy of the MPL was not distributed with this
+  ~ file, You can obtain one at https://mozilla.org/MPL/2.0/.
+-->
+
+# Singleton (Anti)-Pattern
+
+It is important for you to understand the [Singleton (anti)-pattern](https://en.wikipedia.org/wiki/Singleton_pattern) tos
+properly utilize this system.
+
+```admonish help
+Singleton is often referred to as an anti-pattern as a criticism to its use. This is because it violates several good practice
+recommendations for clean, modular code. However it is also a tool that can be used to solve certain design problems. As such,
+it is used internally by Godot, and it is available to tool developers in Godot Rust as well.
+
+[Read more about criticisms here](https://en.wikipedia.org/wiki/Singleton_pattern#Criticism)
+```
+
+An Engine Singleton is registered through [`godot::engine::Engine`](https://godot-rust.github.io/docs/gdext/master/godot/engine/struct.Engine.html).
+
+Custom Engine Singletons in Godot:
+
+- are Object types
+- are always accessible to GDScript (and other available languages like C#)
+- must be manually registered and unregistered in the `LevelInit::Scene` step
+
+Godot does have _many_ built-in singletons provided internally and exposed to GDScript. You can find a full list
+[here](https://docs.godotengine.org/en/stable/classes/class_@globalscope.html#properties)
```

Comment:
Capitalization and backticks for symbols.

Please also use links at the end rather than inline, for readability. The references be moved to the end of the page as well.
```suggestion
An engine singleton is registered through [`godot::engine::Engine`][api-engine].

Custom engine singletons in Godot:

- are `Object` types
- are always accessible to GDScript and GDExtension languages
- must be manually registered and unregistered in the `InitLevel::Scene` step

Godot provides _many_ built-in singletons in its API. You can find a full list [here][godot-singleton-list].

[api-engine]: https://godot-rust.github.io/docs/gdext/master/godot/engine/struct.Engine.html
[godot-singleton-list]: https://docs.godotengine.org/en/stable/classes/class_@globalscope.html#properties
```